Pre_processing_scripts/01_Sort_IT_Extracts.R

```diff
@@ -0,0 +1,129 @@
+rm(list = ls())
+devtools::load_all()
+
+it_extract_path <- file.path(
+  get_slf_dir(),
+  "IT_extracts"
+)
+it_extract_anon_path <- file.path(
+  it_extract_path,
+  "anon-chi-IT"
+)
+
+is_chi_in_file <- function(filename) {
+  data <- read.csv(filename, nrow = 1)
+  return(grepl("UPI", names(data)) %>% any())
+}
+
+# detect task number and tidy up parquet files from last request ---------
+# One can manually delete files and comment this part out.
+get_previous_it_task_num <- function() {
+  existing_parquet_list <- list.files(it_extract_anon_path)
+  ltcs_file <-
+    existing_parquet_list[stringr::str_detect(existing_parquet_list, "LTCs\\.parquet$")]
+  task_num <- stringr::str_extract(ltcs_file, "SCTASK\\d+")
+  if (length(task_num) == 1L) {
+    return(task_num)
+  } else {
+    cli::cli_alert_danger("Detected 0 or more than 1 task numbers! Please check!")
+    return(NULL)
+  }
+}
+
+get_current_it_task_num <- function() {
+  new_csv_files <- list.files(it_extract_path)
+  ltcs_file <-
+    new_csv_files[stringr::str_detect(new_csv_files, "LTCs\\.csv$")]
+  task_num <- stringr::str_extract(ltcs_file, "SCTASK\\d+")
+  if (length(task_num) == 1L) {
+    return(task_num)
+  } else {
+    cli::cli_abort("Detected 0 or more than 1 task numbers! Please check!")
+  }
+}
+
+previous_task_num <- get_previous_it_task_num()
+current_task_num <- get_current_it_task_num()
```

Comment:
```suggestion
new_task_num <- get_current_it_task_num()
```
name change suggestion